packages/@react-spectrum/tag/src/TagGroup.tsx

```diff
@@ -70,6 +72,78 @@ function TagGroup<T extends object>(props: SpectrumTagGroupProps<T>, ref: DOMRef
   }, state, domRef);
   const {tagGroupProps} = useTagGroup(props);
 
+  let [isCollapsed, setIsCollapsed] = useState(maxRows != null);
+  let [visibleTagCount, setVisibleTagCount] = useValueEffect(gridCollection.size);
+
+  let updateVisibleTagCount = useCallback(() => {
+    let computeVisibleTagCount = () => {
+      // Refs can be null at runtime.
+      let currDomRef: HTMLDivElement | null = domRef.current;
+      if (!currDomRef) {
+        return;
+      }
+
+      let tags = [...currDomRef.children].filter(el => el.tagName !== 'BUTTON') as HTMLDivElement[];
+      let currY = -Infinity;
+      let rowCount = 0;
+      let index = 0;
+      // Count rows and show tags until we hit the maxRows.
+      for (let tag of tags) {
+        let {y} = tag.getBoundingClientRect();
+
+        if (y !== currY) {
+          currY = y;
+          rowCount++;
+        }
+
+        if (rowCount > maxRows) {
+          break;
```

Comment:
to do the above comment, you'll have to not `break` since you need the widths of all the remaining tags